### PR TITLE
Add Julia bindings to vm

### PR DESCRIPTION
**Pull request by erikscheurer**
The Julia bindings are released now, this PR adds them to the VM.
Installing Julia uses [jill.py](https://github.com/johnnychen94/jill.py). This is not an official source, but one that allows for installation in a script without manually confirming as well as automatically using the latest Julia release. The [Julialang download site](https://julialang.org/downloads/#long_term_support_release) and the [Julia versioning manager](https://github.com/JuliaLang/juliaup) don't allow for this.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -55,6 +55,7 @@ Vagrant.configure("2") do |config|
   config.vm.provision "shell", path: "provisioning/install-code_aster.sh", privileged: false
   config.vm.provision "shell", path: "provisioning/install-dune.sh", privileged: false
   config.vm.provision "shell", path: "provisioning/install-paraview.sh", privileged: false
+  config.vm.provision "shell", path: "provisioning/install-julia.sh", privileged: false
```
Comment (MakisH) [inline Vagrantfile:58]: This file seems to be missing from the PR.

Comment (erikscheurer) [reply Vagrantfile:58]: Oh right, sorry, without this its a bit pointless. The commit message is also somehow broken, but it should be up to date now.

**Comment by erikscheurer** [comment]
For the Julia bindings we are currently facing [an issue](https://github.com/precice/PreCICE.jl/issues/44) with the installation of the bindings in Ubuntu 22.04. Since we are currently working out a solution, I don't have a permanent fix yet. 
For now, the fix seems to be to build preCICE from source (which we do here already) and then start Julia with preloading: `LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libstdc++.so.6 ./julia` 
Should we already include this in this PR?

**Comment by MakisH** [comment]
> For the Julia bindings we are currently facing [an issue](https://github.com/precice/PreCICE.jl/issues/44) with the installation of the bindings in Ubuntu 22.04. Since we are currently working out a solution, I don't have a permanent fix yet. (And you only notice the error when testing the bindings, which is why I included the commented tests here) For now, the fix seems to be to build preCICE from source (which we do here already) and then start Julia with preloading: `LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libstdc++.so.6 ./julia` Should we already include this in this PR?

We generally don't run tests in this VM, to keep the build time as short as possible (it is already pretty long).

Currently, the VM is still using Ubuntu 20.04 and the upcoming release will also use 20.04. So, nothing to do now, but this comment will be relevant for #45.

I guess we need to set `LD_PRELOAD` every time we start Julia, right? In this case, we could make an `alias` for `julia`, to always set this.

**Comment by MakisH** [comment]
Building currently fails because of the `master`->`main` branch renaming of preCICE. I have fixed that in other branches, but I should probably bring the fix to `develop`. In the meantime, feel free to rename it yourself and trigger the build by marking the PR as draft and then ready for review again.

**Comment by MakisH** [comment]
Building currently seems to fail at downloading code-aster (unrelated), due to an SSL certificate issue.

**Comment by erikscheurer** [comment]
> Building currently seems to fail at downloading code-aster (unrelated), due to an SSL certificate issue.

Is this something we can fix in this PR?

**Comment by MakisH** [comment]
> > Building currently seems to fail at downloading code-aster (unrelated), due to an SSL certificate issue.
> 
> Is this something we can fix in this PR?

It is not related to this PR or anything from our side. I can work around this issue (insecurely), but I will give it some time, hoping for an upstream solution.

**Comment by MakisH** [comment]
@erikscheurer can you please give me push access to your fork? You can enable this with a checkbox in the side of this PR.

**Review by MakisH — APPROVED** [review]

**Pull request merged**